lib/Driver/ToolChains.cpp

```diff
@@ -1101,33 +1101,49 @@ toolchains::GenericUnix::constructInvocation(const LinkJobAction &job,
 
   ArgStringList Arguments;
 
+  // Explicitly pass the target to the linker
+  Arguments.push_back(context.Args.MakeArgString("--target=" + getTriple().str()));
```

Comment:
Er, I see it's already like this, okay. Why move this up here from where it was?